### PR TITLE
Use API price, and refetch logic

### PR DESCRIPTION
**Pull request by anxolin**
This PR consolidates 3 PRs:
* Refactor operator get operation and add get price (described below)
* Price quote/refactor state #601
* [state] Get price from our API #602

# Description
Adds the new logic for fetching the price from the API, and reacting to changes of the amount/tokens. Also updates the price and fee periodically. The price is updating more frequently, and the fee uses the expiration date.

# Refactor operator get operation and add get price
First of several PRs for including our own price estimation in CowSwap

## Scope
This one includes a refactor of the operator HTTP get methods removing duplicated logic. Also, incorporates the method for getting a quote from the API https://protocol-rinkeby.dev.gnosisdev.com/api/#/

## Test
For now, let's just test that this PR doesn't break the fee estimation.

1. Select token pair
2. Input amount
3. Check the network, verify the request for the fee was done, and that is shown in the UI

------

### CODE REVIEW

**Review by W3stside — APPROVED** [review]
everything good in networks tab and fees work, approve

File: src/custom/utils/operator.ts [hunk 1]
```diff
@@ -198,20 +203,10 @@ function toApiAddress(address: string, chainId: ChainId): string {
   return address
 }
 
-export async function getFeeQuote(params: FeeQuoteParams): Promise<FeeInformation> {
-  const { sellToken, buyToken, amount, kind, chainId } = params
-  const [checkedSellAddress, checkedBuyAddress] = [checkIfEther(sellToken, chainId), checkIfEther(buyToken, chainId)]
-  console.log('[util:operator] Get fee from API', params)
-
+async function _getJson(chainId: ChainId, url: string): Promise<any> {
```
Comment (alfetopito) [inline src/custom/utils/operator.ts:206]: Minor: what about making it generic, to get rid of that pesky `any`?

```suggestion
async function _getJson<T>(chainId: ChainId, url: string): Promise<T> {
```

And when calling it you might need to say the type, like `FeeInformation`, etc:

**Review by alfetopito — APPROVED** [review]
Working fine.

Minor comment about generics ;)

File: src/custom/utils/operator.ts (hunk 1 above)
Comment (anxolin) [reply src/custom/utils/operator.ts:206]: Okay, fine with me. I thought about doing it, but passed cause:
- fetch returns `Promise<any>` and that's just a wrapper of fetch. 
- It work automatically without casting, it inferred ok the types, so I left this approach cause i don't need to add the generic information

Happy to add it too though, let me know if i miss sth

Comment (alfetopito) [reply src/custom/utils/operator.ts:206]: :shrug:

**Pull request merged**